src/js/containers/search/visualizations/rank/SpendingByRecipientVisualizationContainer.jsx

```diff
@@ -0,0 +1,276 @@
+            next: '',
+            previous: '',
+            hasNextPage: false,
+            hasPreviousPage: false
+        };
+
+        this.changeScope = this.changeScope.bind(this);
+        this.nextPage = this.nextPage.bind(this);
+        this.previousPage = this.previousPage.bind(this);
+        this.apiRequest = null;
+    }
+
+    componentDidMount() {
+        this.newSearch();
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!_.isEqual(prevProps.reduxFilters, this.props.reduxFilters)
+            || (prevProps.budgetFiltersSelected !== this.props.budgetFiltersSelected)
+            || (prevProps.awardFiltersSelected !== this.props.awardFiltersSelected)) {
+            this.newSearch();
+        }
+    }
+
+    changeScope(scope) {
+        this.setState({
+            scope,
+            page: 1,
+            hasNextPage: false
+        }, () => {
+            this.fetchData();
+        });
+    }
+
+    newSearch() {
+        this.setState({
+            page: 1,
+            hasNextPage: false
+        }, () => {
+            this.fetchData();
+        });
+    }
+
+    nextPage() {
+        if (this.state.hasNextPage) {
+            this.setState({
+                page: this.state.page + 1
+            }, () => {
+                this.fetchData();
+            });
+        }
+    }
+
+    previousPage() {
+        // change the state by subtracting 2 (since the page number is already incremented)
+        const prevPage = _.max([1, this.state.page - 1]);
+        this.setState({
+            page: prevPage
+        }, () => {
+            this.fetchData();
+        });
+    }
+
+    fetchData() {
+        this.setState({
+            loading: true
+        });
+
+        // Cancel API request if it exists
+        if (this.apiRequest) {
+            this.apiRequest.cancel();
+        }
+
+        // Fetch data from the appropriate endpoint
+        if (this.props.awardFiltersSelected && this.props.budgetFiltersSelected) {
+            this.fetchComboRequest();
+        }
+        else if (this.props.budgetFiltersSelected) {
+            this.fetchBudgetRequest();
+        }
+        else if (this.props.awardFiltersSelected) {
+            this.fetchAwardRequest();
+        }
+        else {
+            this.fetchUnfilteredRequest();
+        }
+    }
+
+    fetchUnfilteredRequest() {
+        this.fetchTransactions('Recipient vis - unfiltered');
+    }
+
+    fetchBudgetRequest() {
+        this.fetchAccountAwards('Recipient vis - budget filters');
+    }
+
+    fetchAwardRequest() {
+        // only award filters have been selected
+        this.fetchTransactions('Recipient vis - award filters');
+    }
+
+    fetchComboRequest() {
+        // a combination of budget and award filters have been selected
+        this.fetchAccountAwards('Recipient vis - combination');
+    }
+
+    fetchTransactions(auditTrail = null) {
+        const idField = FilterFields.accountAwardsFields.recipientId;
```

Comment:
Why are you pulling from `accountAwardsFields` in the transactions function? Why group by `award__recipient__xxx` instead of `recipient__xxx`?